Agent node hangs for over 10 minutes and eventually times out on second question, Docker logs show "bufio.Scanner: token too long" error

### Self Checks

- [x] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [x] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:)
- [x] Please do not modify this template :) and fill in all the required fields.

### Dify version

1.3.1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

![Image](https://github.com/user-attachments/assets/3c1bee9c-e47a-4476-9a27-3e30f3c799c7)

![Image](https://github.com/user-attachments/assets/bb784a35-bfa1-4f88-9972-20bac1179cb0)

I observed a issue when using the langgenius/agent:0.0.15 plugin in a simple ChatFlow workflow:
Start Node → Agent Node (Function Calling strategy, one tool, short prompt) → Answer Node

Steps & Results:
Step 1: Upload a ~4MB spreadsheet file and ask a simple question like “What time is it?”
✅ Result: Agent responds correctly within seconds.
Step 2: Immediately ask the same question again.
❌ Result: Agent enters a "running" state and remains unresponsive for over 10 minutes , finally failing with:
`Run failed: PluginDaemonInternalServerError: killed by timeout`

During this period, the following error appears in the Docker logs:
`[ERROR] plugin langgenius/agent:0.0.15 has an error on stdout: bufio.Scanner: token too long`

Eventually, the plugin process is killed:
`[ERROR] plugin langgenius/agent:0.0.15 exited with error: signal: killed`

Below is the relevant Docker log output:
```text
nginx         | 2025-05-10 21:46:02.133 | 172.19.0.1 - - [10/May/2025:13:46:02 +0000] "GET /console/api/workspaces/current/models/model-types/text-embedding HTTP/1.1" 200 3923 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:46:02.133 | 172.19.0.1 - - [10/May/2025:13:46:02 +0000] "GET /console/api/workspaces/current/models/model-types/llm HTTP/1.1" 200 59084 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:46:02.149 | 172.19.0.1 - - [10/May/2025:13:46:02 +0000] "GET /console/api/workspaces/current/models/model-types/speech2text HTTP/1.1" 200 2816 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:46:02.164 | 172.19.0.1 - - [10/May/2025:13:46:02 +0000] "GET /console/api/workspaces/current/model-providers/langgenius/deepseek/deepseek/models HTTP/1.1" 200 3505 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
plugin_daemon | 2025-05-10 21:46:02.976 | [GIN] 2025/05/10 - 13:46:02 | 200 |   13.428388ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
nginx         | 2025-05-10 21:46:03.094 | 172.19.0.1 - - [10/May/2025:13:46:03 +0000] "GET /console/api/workspaces/current/plugin/list HTTP/1.1" 200 1040469 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:46:06.923 | 172.19.0.1 - - [10/May/2025:13:46:06 +0000] "POST /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 124 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:46:10.160 | 2025/05/10 13:46:10 [warn] 36#36: *185 a client request body is buffered to a temporary file /var/cache/nginx/client_temp/0000000001, client: 172.19.0.1, server: _, request: "POST /console/api/files/upload HTTP/1.1", host: "localhost", referrer: "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow"
nginx         | 2025-05-10 21:46:10.430 | 172.19.0.1 - - [10/May/2025:13:46:10 +0000] "POST /console/api/files/upload HTTP/1.1" 201 297 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:46:18.979 | 172.19.0.1 - - [10/May/2025:13:46:18 +0000] "POST /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 124 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
plugin_daemon | 2025-05-10 21:46:19.576 | [GIN] 2025/05/10 - 13:46:19 | 200 |   15.019587ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:19.607 | [GIN] 2025/05/10 - 13:46:19 | 200 |    1.401759ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/agent_strategy?provider=agent&plugin_id=langgenius%2Fagent"
plugin_daemon | 2025-05-10 21:46:19.630 | [GIN] 2025/05/10 - 13:46:19 | 200 |    9.666289ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/models?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:20.738 | [GIN] 2025/05/10 - 13:46:20 | 200 |  1.064649114s |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/dispatch/model/schema"
plugin_daemon | 2025-05-10 21:46:20.766 | [GIN] 2025/05/10 - 13:46:20 | 200 |   14.320678ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:22.085 | [GIN] 2025/05/10 - 13:46:22 | 200 |   15.948075ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:22.116 | [GIN] 2025/05/10 - 13:46:22 | 200 |   11.604146ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/models?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:22.144 | [GIN] 2025/05/10 - 13:46:22 | 200 |    2.234029ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/tools?page=1&page_size=256"
sandbox       | 2025-05-10 21:46:23.052 | [GIN] 2025/05/10 - 13:46:23 | 200 |     933.963µs |             ::1 | GET      "/health"
plugin_daemon | 2025-05-10 21:46:28.107 | [GIN] 2025/05/10 - 13:46:28 | 200 |  5.693080657s |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/dispatch/llm/invoke"
plugin_daemon | 2025-05-10 21:46:28.140 | [GIN] 2025/05/10 - 13:46:28 | 200 |   13.097887ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.170 | [GIN] 2025/05/10 - 13:46:28 | 200 |   10.714246ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/models?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.202 | [GIN] 2025/05/10 - 13:46:28 | 200 |     382.632µs |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/tools?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.221 | [GIN] 2025/05/10 - 13:46:28 | 200 |   13.330689ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.276 | [GIN] 2025/05/10 - 13:46:28 | 200 |     497.773µs |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/tools?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.297 | [GIN] 2025/05/10 - 13:46:28 | 200 |    13.07392ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.352 | [GIN] 2025/05/10 - 13:46:28 | 200 |      534.68µs |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/tools?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.373 | [GIN] 2025/05/10 - 13:46:28 | 200 |   14.140634ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:28.437 | [GIN] 2025/05/10 - 13:46:28 | 200 |     577.765µs |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/tools?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:32.736 | [GIN] 2025/05/10 - 13:46:32 | 200 |  4.538393394s |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/dispatch/llm/invoke"
plugin_daemon | 2025-05-10 21:46:32.743 | [GIN] 2025/05/10 - 13:46:32 | 200 | 11.647517718s |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/dispatch/agent_strategy/invoke"
plugin_daemon | 2025-05-10 21:46:32.755 | [GIN] 2025/05/10 - 13:46:32 | 200 |    13.15063ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:32.811 | [GIN] 2025/05/10 - 13:46:32 | 200 |     505.906µs |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/tools?page=1&page_size=256"
nginx         | 2025-05-10 21:46:32.871 | 172.19.0.1 - - [10/May/2025:13:46:32 +0000] "POST /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/advanced-chat/workflows/draft/run HTTP/1.1" 200 27971 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:46:33.355 | 172.19.0.1 - - [10/May/2025:13:46:33 +0000] "POST /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 124 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
plugin_daemon | 2025-05-10 21:46:33.447 | [GIN] 2025/05/10 - 13:46:33 | 200 |   12.250669ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:33.477 | [GIN] 2025/05/10 - 13:46:33 | 200 |       625.2µs |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/agent_strategy?provider=agent&plugin_id=langgenius%2Fagent"
plugin_daemon | 2025-05-10 21:46:33.495 | [GIN] 2025/05/10 - 13:46:33 | 200 |    9.252196ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/models?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:33.544 | [GIN] 2025/05/10 - 13:46:33 | 200 |    4.319266ms |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/dispatch/model/schema"
plugin_daemon | 2025-05-10 21:46:33.672 | [GIN] 2025/05/10 - 13:46:33 | 200 |    13.80995ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
sandbox       | 2025-05-10 21:46:53.081 | [GIN] 2025/05/10 - 13:46:53 | 200 |      34.509µs |             ::1 | GET      "/health"
plugin_daemon | 2025-05-10 21:46:55.408 | [GIN] 2025/05/10 - 13:46:55 | 200 |   14.390146ms |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/list?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:46:55.438 | 2025/05/10 13:46:55 stdio.go:188: [ERROR]plugin langgenius/agent:0.0.15 has an error on stdout: bufio.Scanner: token too long
plugin_daemon | 2025-05-10 21:46:55.459 | 2025/05/10 13:46:55 run.go:135: [ERROR]plugin langgenius/agent:0.0.15 exited with error: signal: killed
plugin_daemon | 2025-05-10 21:46:55.461 | 2025/05/10 13:46:55 run.go:190: [INFO]plugin langgenius/agent:0.0.15 stopped
plugin_daemon | 2025-05-10 21:46:55.475 | [GIN] 2025/05/10 - 13:46:55 | 200 |     482.534µs |      172.19.0.9 | GET      "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/tools?page=1&page_size=256"
plugin_daemon | 2025-05-10 21:47:00.466 | 2025/05/10 13:47:00 run.go:147: [INFO]plugin langgenius/agent:0.0.15 started
plugin_daemon | 2025-05-10 21:47:06.586 | 2025/05/10 13:47:06 stdio.go:182: [INFO]plugin langgenius/agent:0.0.15: Installed agent: agent
plugin_daemon | 2025-05-10 21:47:23.013 | [GIN] 2025/05/10 - 13:47:23 | 200 |     2.41423ms |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/installation/fetch/batch"
nginx         | 2025-05-10 21:47:23.031 | 172.19.0.1 - - [10/May/2025:13:47:23 +0000] "POST /console/api/workspaces/current/plugin/list/installations/ids HTTP/1.1" 200 2119 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
sandbox       | 2025-05-10 21:47:23.112 | [GIN] 2025/05/10 - 13:47:23 | 200 |      78.904µs |             ::1 | GET      "/health"
nginx         | 2025-05-10 21:47:23.501 | 172.19.0.1 - - [10/May/2025:13:47:23 +0000] "GET /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 8211 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
sandbox       | 2025-05-10 21:47:53.125 | [GIN] 2025/05/10 - 13:47:53 | 200 |     496.072µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:48:23.152 | [GIN] 2025/05/10 - 13:48:23 | 200 |      26.239µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:48:53.188 | [GIN] 2025/05/10 - 13:48:53 | 200 |     355.306µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:49:23.219 | [GIN] 2025/05/10 - 13:49:23 | 200 |      33.416µs |             ::1 | GET      "/health"
redis         | 2025-05-10 21:49:53.047 | 1:M 10 May 2025 13:49:53.046 * 100 changes in 300 seconds. Saving...
redis         | 2025-05-10 21:49:53.048 | 1:M 10 May 2025 13:49:53.048 * Background saving started by pid 74
redis         | 2025-05-10 21:49:53.070 | 74:C 10 May 2025 13:49:53.070 * DB saved on disk
redis         | 2025-05-10 21:49:53.070 | 74:C 10 May 2025 13:49:53.070 * RDB: 0 MB of memory used by copy-on-write
redis         | 2025-05-10 21:49:53.149 | 1:M 10 May 2025 13:49:53.149 * Background saving terminated with success
sandbox       | 2025-05-10 21:49:53.247 | [GIN] 2025/05/10 - 13:49:53 | 200 |      26.411µs |             ::1 | GET      "/health"
db            | 2025-05-10 21:49:54.590 | 2025-05-10 13:49:54.589 UTC [33] LOG:  checkpoint starting: time
db            | 2025-05-10 21:50:01.878 | 2025-05-10 13:50:01.878 UTC [33] LOG:  checkpoint complete: wrote 74 buffers (0.5%); 0 WAL file(s) added, 0 removed, 0 recycled; write=7.220 s, sync=0.030 s, total=7.290 s; sync files=46, longest=0.003 s, average=0.001 s; distance=391 kB, estimate=391 kB
sandbox       | 2025-05-10 21:50:23.277 | [GIN] 2025/05/10 - 13:50:23 | 200 |      38.605µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:50:53.306 | [GIN] 2025/05/10 - 13:50:53 | 200 |     102.387µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:51:23.331 | [GIN] 2025/05/10 - 13:51:23 | 200 |       41.26µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:51:53.361 | [GIN] 2025/05/10 - 13:51:53 | 200 |      47.091µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:52:23.388 | [GIN] 2025/05/10 - 13:52:23 | 200 |      25.019µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:52:53.418 | [GIN] 2025/05/10 - 13:52:53 | 200 |      24.297µs |             ::1 | GET      "/health"
plugin_daemon | 2025-05-10 21:53:09.479 | [GIN] 2025/05/10 - 13:53:09 | 200 |    1.993259ms |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/installation/fetch/batch"
nginx         | 2025-05-10 21:53:09.482 | 172.19.0.1 - - [10/May/2025:13:53:09 +0000] "POST /console/api/workspaces/current/plugin/list/installations/ids HTTP/1.1" 200 2119 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:53:09.942 | 172.19.0.1 - - [10/May/2025:13:53:09 +0000] "GET /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 8211 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
sandbox       | 2025-05-10 21:53:23.444 | [GIN] 2025/05/10 - 13:53:23 | 200 |      47.863µs |             ::1 | GET      "/health"
plugin_daemon | 2025-05-10 21:53:29.474 | [GIN] 2025/05/10 - 13:53:29 | 200 |    1.199121ms |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/installation/fetch/batch"
nginx         | 2025-05-10 21:53:29.476 | 172.19.0.1 - - [10/May/2025:13:53:29 +0000] "POST /console/api/workspaces/current/plugin/list/installations/ids HTTP/1.1" 200 2119 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:53:29.931 | 172.19.0.1 - - [10/May/2025:13:53:29 +0000] "GET /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 8211 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
sandbox       | 2025-05-10 21:53:53.471 | [GIN] 2025/05/10 - 13:53:53 | 200 |      25.269µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:54:23.498 | [GIN] 2025/05/10 - 13:54:23 | 200 |      25.189µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:54:53.530 | [GIN] 2025/05/10 - 13:54:53 | 200 |     347.593µs |             ::1 | GET      "/health"
redis         | 2025-05-10 21:54:54.062 | 1:M 10 May 2025 13:54:54.062 * 100 changes in 300 seconds. Saving...
redis         | 2025-05-10 21:54:54.063 | 1:M 10 May 2025 13:54:54.062 * Background saving started by pid 141
redis         | 2025-05-10 21:54:54.074 | 141:C 10 May 2025 13:54:54.073 * DB saved on disk
redis         | 2025-05-10 21:54:54.074 | 141:C 10 May 2025 13:54:54.074 * RDB: 0 MB of memory used by copy-on-write
redis         | 2025-05-10 21:54:54.163 | 1:M 10 May 2025 13:54:54.163 * Background saving terminated with success
db            | 2025-05-10 21:54:54.925 | 2025-05-10 13:54:54.924 UTC [33] LOG:  checkpoint starting: time
db            | 2025-05-10 21:54:55.056 | 2025-05-10 13:54:55.056 UTC [33] LOG:  checkpoint complete: wrote 2 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.108 s, sync=0.003 s, total=0.132 s; sync files=2, longest=0.002 s, average=0.002 s; distance=4 kB, estimate=353 kB
sandbox       | 2025-05-10 21:55:23.562 | [GIN] 2025/05/10 - 13:55:23 | 200 |     445.865µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:55:53.633 | [GIN] 2025/05/10 - 13:55:53 | 200 |      26.262µs |             ::1 | GET      "/health"
plugin_daemon | 2025-05-10 21:56:11.977 | [GIN] 2025/05/10 - 13:56:11 | 200 |    1.607913ms |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/installation/fetch/batch"
nginx         | 2025-05-10 21:56:12.061 | 172.19.0.1 - - [10/May/2025:13:56:12 +0000] "POST /console/api/workspaces/current/plugin/list/installations/ids HTTP/1.1" 200 2119 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:56:12.372 | 172.19.0.1 - - [10/May/2025:13:56:12 +0000] "GET /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 8211 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
sandbox       | 2025-05-10 21:56:23.667 | [GIN] 2025/05/10 - 13:56:23 | 200 |      43.324µs |             ::1 | GET      "/health"
sandbox       | 2025-05-10 21:56:53.697 | [GIN] 2025/05/10 - 13:56:53 | 200 |      34.927µs |             ::1 | GET      "/health"
plugin_daemon | 2025-05-10 21:56:54.943 | 2025/05/10 13:56:54 factory.go:28: [ERROR]PluginDaemonInternalServerError: killed by timeout
plugin_daemon | 2025-05-10 21:56:54.943 | goroutine 338 [running]:
plugin_daemon | 2025-05-10 21:56:54.943 | runtime/debug.Stack()
plugin_daemon | 2025-05-10 21:56:54.943 | 	/usr/local/go/src/runtime/debug/stack.go:26 +0x5e
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/types/exception.InternalServerError({0x2717460, 0xc000923570})
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/types/exception/factory.go:27 +0x27
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/service.baseSSEService[...](0xc0038950c8, 0x2425cad, 0x258?)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/service/base_sse.go:84 +0x4c6
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/service.baseSSEWithSession[...](0xc003895158, {0x2425cad, 0xe}, {0x24351db, 0x15}, 0xc00076c6c0, 0xc004256200, 0x258)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/service/base_sse.go:114 +0x1c5
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/service.InvokeAgentStrategy(0x100c0002ba510?, 0x21ac120?, 0xc001210450?)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/service/invoke_agent.go:19 +0x52
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server.(*App).pluginDispatchGroup.InvokeAgentStrategy.func7.1({{{0xc00312550d, 0x24}, {0xc00419d620, 0x24}}, {{0x0, 0x0}}, {0xc003125560, 0x58}, 0xc001210490, 0x0, ...})
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/controllers/agent.go:20 +0x91
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server/controllers.BindPluginDispatchRequest[...].func1()
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/controllers/base.go:53 +0xb5
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server/controllers.BindRequest[...](0xc004256200, 0xc0038953a0)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/controllers/base.go:31 +0x275
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server/controllers.BindPluginDispatchRequest[...](0x241eb5e?, 0xa?)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/controllers/base.go:37 +0x36
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server.(*App).pluginDispatchGroup.InvokeAgentStrategy.func7(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/controllers/agent.go:17 +0x74
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Context).Next(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x2b
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server.(*App).pluginDispatchGroup.(*App).InitClusterID.func3(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/middleware.go:170 +0x54
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Context).Next(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x2b
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server.(*App).pluginDispatchGroup.(*App).RedirectPluginInvoke.func2(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/middleware.go:97 +0x74
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Context).Next(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x2b
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server.(*App).pluginDispatchGroup.(*App).FetchPluginInstallation.func1(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/middleware.go:66 +0x2d2
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Context).Next(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185 +0x2b
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/langgenius/dify-plugin-daemon/internal/server.(*App).pluginGroup.CheckingKey.func1(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/app/internal/server/middleware.go:24 +0x95
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Context).Next(...)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.CustomRecoveryWithWriter.func1(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/recovery.go:102 +0x6f
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Context).Next(...)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.LoggerWithConfig.func1(0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/logger.go:249 +0xe5
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Context).Next(...)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/context.go:185
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0xc0002456c0, 0xc004256200)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:633 +0x892
plugin_daemon | 2025-05-10 21:56:54.943 | github.com/gin-gonic/gin.(*Engine).ServeHTTP(0xc0002456c0, {0x2744ff0, 0xc00400c2a0}, 0xc0000fd2c0)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/go/pkg/mod/github.com/gin-gonic/gin@v1.10.0/gin.go:589 +0x1b2
plugin_daemon | 2025-05-10 21:56:54.943 | net/http.serverHandler.ServeHTTP({0x2738308?}, {0x2744ff0?, 0xc00400c2a0?}, 0x6?)
plugin_daemon | 2025-05-10 21:56:54.943 | 	/usr/local/go/src/net/http/server.go:3210 +0x8e
plugin_daemon | 2025-05-10 21:56:54.943 | net/http.(*conn).serve(0xc004294510, {0x2748a90, 0xc000920900})
plugin_daemon | 2025-05-10 21:56:54.943 | 	/usr/local/go/src/net/http/server.go:2092 +0x5d0
plugin_daemon | 2025-05-10 21:56:54.943 | created by net/http.(*Server).Serve in goroutine 63
plugin_daemon | 2025-05-10 21:56:54.943 | 	/usr/local/go/src/net/http/server.go:3360 +0x485
plugin_daemon | 2025-05-10 21:56:54.943 | 
plugin_daemon | 2025-05-10 21:56:54.944 | [GIN] 2025/05/10 - 13:56:54 | 200 |        10m21s |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/dispatch/agent_strategy/invoke"
api           | 2025-05-10 21:56:54.965 | 2025-05-10 13:56:54.945 ERROR [Thread-5 (_generate_worker)] [graph_engine.py:844] - Node Agent run failed
api           | 2025-05-10 21:56:54.966 | Traceback (most recent call last):
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/workflow/graph_engine/graph_engine.py", line 647, in _run_node
api           | 2025-05-10 21:56:54.966 |     for item in generator:
api           | 2025-05-10 21:56:54.966 |                 ^^^^^^^^^
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/workflow/nodes/base/node.py", line 84, in run
api           | 2025-05-10 21:56:54.966 |     yield from result
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/workflow/nodes/agent/agent_node.py", line 99, in _run
api           | 2025-05-10 21:56:54.966 |     yield from self._transform_message(
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/workflow/nodes/tool/tool_node.py", line 208, in _transform_message
api           | 2025-05-10 21:56:54.966 |     for message in message_stream:
api           | 2025-05-10 21:56:54.966 |                    ^^^^^^^^^^^^^^
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/tools/utils/message_transformer.py", line 25, in transform_tool_invoke_messages
api           | 2025-05-10 21:56:54.966 |     for message in messages:
api           | 2025-05-10 21:56:54.966 |                    ^^^^^^^^
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/agent/strategy/base.py", line 25, in invoke
api           | 2025-05-10 21:56:54.966 |     yield from self._invoke(params, user_id, conversation_id, app_id, message_id)
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/agent/strategy/plugin.py", line 50, in _invoke
api           | 2025-05-10 21:56:54.966 |     yield from manager.invoke(
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/plugin/impl/base.py", line 191, in _request_with_plugin_daemon_response_stream
api           | 2025-05-10 21:56:54.966 |     self._handle_plugin_daemon_error(error.error_type, error.message)
api           | 2025-05-10 21:56:54.966 |   File "/app/api/core/plugin/impl/base.py", line 225, in _handle_plugin_daemon_error
api           | 2025-05-10 21:56:54.966 |     raise PluginDaemonInternalServerError(description=message)
api           | 2025-05-10 21:56:54.966 | core.plugin.impl.exc.PluginDaemonInternalServerError: PluginDaemonInternalServerError: killed by timeout
nginx         | 2025-05-10 21:56:55.030 | 172.19.0.1 - - [10/May/2025:13:56:55 +0000] "POST /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/advanced-chat/workflows/draft/run HTTP/1.1" 200 8233 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
plugin_daemon | 2025-05-10 21:57:21.866 | [GIN] 2025/05/10 - 13:57:21 | 200 |     980.321µs |      172.19.0.9 | POST     "/plugin/d715e894-8b1c-46e0-9c7d-d721fc67e0f1/management/installation/fetch/batch"
nginx         | 2025-05-10 21:57:21.868 | 172.19.0.1 - - [10/May/2025:13:57:21 +0000] "POST /console/api/workspaces/current/plugin/list/installations/ids HTTP/1.1" 200 2119 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
nginx         | 2025-05-10 21:57:22.380 | 172.19.0.1 - - [10/May/2025:13:57:22 +0000] "GET /console/api/apps/8308a8a0-c311-4678-8169-f102360a5001/workflows/draft HTTP/1.1" 200 8211 "http://localhost/app/8308a8a0-c311-4678-8169-f102360a5001/workflow" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/136.0.0.0 Safari/537.36 Edg/136.0.0.0" "-"
```

### ✔️ Expected Behavior

The agent should respond quickly or fail fast after encountering an error — not hang for more than 10 minutes.

### ❌ Actual Behavior

First question works normally.
Second question gets stuck in "running" state for ~10+ minutes before timing out.
Error log indicates that bufio.Scanner fails due to a line exceeding its maximum buffer size.

--- a/api/core/plugin/backwards_invocation/model.py
+++ b/api/core/plugin/backwards_invocation/model.py
@@ -58,6 +58,7 @@ class PluginModelBackwardsInvocation(BaseBackwardsInvocation):
                         LLMNode.deduct_llm_quota(
                             tenant_id=tenant.id, model_instance=model_instance, usage=chunk.delta.usage
                         )
+                    chunk.prompt_messages = []
                     yield chunk
 
             return handle()
@@ -68,7 +69,7 @@ class PluginModelBackwardsInvocation(BaseBackwardsInvocation):
             def handle_non_streaming(response: LLMResult) -> Generator[LLMResultChunk, None, None]:
                 yield LLMResultChunk(
                     model=response.model,
-                    prompt_messages=response.prompt_messages,
+                    prompt_messages=[],
                     system_fingerprint=response.system_fingerprint,
                     delta=LLMResultChunkDelta(
                         index=0,

